de.fhdo.lemma.service.open_api/src/de/fhdo/lemma/service/open_api/LemmaDataSubGenerator.xtend

```diff
@@ -0,0 +1,197 @@
+        } else
+            throw new Exception("Data model generation failed :(")
+        return myDataModel
+    }
+
+    def initilize() {
+        // Data Model
+        myDataModel.versions.add(myVersion)
+        // Version
+        myVersion.name = OpenApiUtil.removeInvalidCharsFromName(openAPI.info.version)
+        myVersion.contexts.add(myContext)
+        // Context
+        myContext.name = OpenApiUtil.removeInvalidCharsFromName(openAPI.info.title)
+        myContext.version = myVersion
+    }
+
+    def DataStructure getOrCreateDataStructure(
+        Context c, String name, Schema schema
+    ) {
+        var foundObject = findDataStructure(c, StringUtils.capitalize(name))
+        if (foundObject === null) {
+            val newDataStructure = DATA_FACTORY.createDataStructure
+            newDataStructure.name = StringUtils.capitalize(name)
+            c.complexTypes.add(newDataStructure)
+            addDataFieldsToDataStructure(newDataStructure, name, schema)
+            createdDataStructures.put(
+                newDataStructure.buildQualifiedName(seperator), newDataStructure
+            )
+            return newDataStructure
+        } else {
+            return foundObject
+        }
+    }
+
+    def ListType getOrCreateListStructure(Context c, String name, Schema schema) {
+        var foundObject = findListStructure(c, name)
+        if (foundObject === null) {
+            val newListStructure = DATA_FACTORY.createListType
+            newListStructure.name = '''«StringUtils.capitalize(name)»List'''
+            c.complexTypes.add(newListStructure)
+            addDataFieldsToListStructure(newListStructure, name, schema)
+            createdListStructures.put(
+                newListStructure.buildQualifiedName(seperator), newListStructure
+            )
+            return newListStructure
+        } else {
+            return foundObject
+        }
+    }
+
+    def void addDataFieldsToDataStructure(
+        DataStructure ds, String structureName, Schema structureSchema
+    ) {
+        if (structureSchema.properties !== null) {
+            structureSchema.properties.forEach[
+                name, schema|addDataFieldsToDataStructure(ds, name, schema)
+            ]
+        } else {
+            val newDataField = generateDataField(structureName, structureSchema)
+            ds.dataFields.add(newDataField)
+        }
+    }
+
+    def addDataFieldsToListStructure(
+        ListType ds, String structureName, Schema structureSchema
+    ) {
+        val newDataField = generateDataField(structureName, structureSchema)
+        ds.dataFields.add(newDataField)
+    }
+
+    def DataField generateDataField(
+        String structureName, Schema structureSchema
+    ) {
+        val newDataField = DATA_FACTORY.createDataField
+        newDataField.name = StringUtils.lowerCase(structureName)
+
+        switch (structureSchema.type) {
+            case "integer": {
```

Comment:
Use fall-through cases for this and the next two case expressions (https://www.eclipse.org/xtend/documentation/203_xtend_expressions.html#switch-expression). Also, remove the redundant curly brackets then.